qiskit/providers/backend.py

```diff
@@ -229,3 +233,233 @@ class can handle either situation.
+        self._options = self._default_options()
+        self._provider = provider
+        self._coupling_map = None
+        if fields:
+            for field in fields:
+                if field not in self._options.data:
+                    raise AttributeError("Options field %s is not valid for this backend" % field)
+            self._options.update_config(**fields)
+
+    @property
+    def instructions(self) -> List[Gate]:
+        """A list of Gate instances that the backend supports."""
+        return list(self.target.gates)
+
+    @property
+    def instructions_names(self) -> List[str]:
+        """A list of Gate names that the backend supports."""
+        return list(self.target.gate_names)
+
+    @property
+    @abstractmethod
+    def target(self):
+        """A :class:`qiskit.transpiler.Target` object for the backend."""
+        pass
+
+    @property
+    @abstractmethod
+    def num_qubits(self) -> int:
+        """Return the number of qubits the backend has."""
+        pass
+
+    @property
+    def coupling_map(self):
+        """Return the :class:`~qiskit.transpiler.CouplingMap` object"""
+        return self.target.coupling_map()
+
+    @property
+    def instruction_durations(self):
+        """Return the :class:`~qiskit.transpiler.InstructionDurations` object."""
+        return self.target.durations()
+
+    @property
+    @abstractmethod
+    def conditional(self) -> bool:
+        """Return bool whether the target can execute gates with classical
+        conditions."""
+        pass
+
+    @property
+    @abstractmethod
+    def max_shots(self) -> int:
+        """Return the maximum number of shots supported by the backend.
+
+        If there is no limit this will return None
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def max_experiments(self):
+        """The maximum number of circuits (or Pulse schedules) that can be
+        run in a single job.
+
+        If there is no limit this will return None
+        """
+        pass
+
+    @classmethod
+    @abstractmethod
+    def _default_options(cls):
+        """Return the default options
+
+        This method will return a :class:`qiskit.providers.Options`
+        subclass object that will be used for the default options. These
+        should be the default parameters to use for the options of the
+        backend.
+
+        Returns:
+            qiskit.providers.Options: A options object with
+                default values set
+        """
+        pass
+
+    def t1(self, qubit: int):
+        """Return the T1 time of a given qubit
+
+        Args:
+            qubit: The qubit index to get the T1 time for
+
+        Returns:
+            t1: the T1 time for the specified qubit in seconds
+
+        Raises:
+            NotImplementedError: if the backend doesn't support querying the
+                t1 time for a qubit
+        """
+        raise NotImplementedError
+
+    def t2(self, qubit: int):
```

Comment:
So what one often wants to do is not just get the T1 or T2 time of a single qubit, but of a collection of qubits, and then ask things like what is the mean, median, max, min, etc.  This is even more important as systems get larger and one has to deal with them in pieces.  It would be nice if things could return a NumPy array of values that can be sliced up.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/ad7bbf5890b3b8c8ee2d85304584215a4c519d7d

Reply:
I think I have never once needed a single qubit T1 or T2 value.  If I did I probably just looked at the IQX.  Can it be a numpy array.  That way it can be easily sliced or things like `t2_array.mean()` also work.  Alternatively it could be a class that allows the calls you have, plus a `_getitem_` that could return a sliced array.

Reply:
Sure, I'll change it to a numpy array return for a list input.